### PR TITLE
DDS Fix doxygen [8422]

### PR DESCRIPTION
**Pull request by laura-eprosima**

### CODE REVIEW

File: include/fastdds/dds/core/status/DeadlineMissedStatus.hpp [hunk 1]
```diff
@@ -51,7 +53,9 @@ struct DeadlineMissedStatus
     fastrtps::rtps::InstanceHandle_t last_instance_handle;
 };
 
+//! Alias of DeadlineMissedStatus
 typedef DeadlineMissedStatus OfferedDeadlineMissedStatus;
+//! Alias of DeadlineMissedStatus
```
Comment (MiguelCompany) [inline include/fastdds/dds/core/status/DeadlineMissedStatus.hpp:58]: We do aliases with `using`

File: include/fastdds/dds/core/status/SampleRejectedStatus.hpp [hunk 2]
```diff
@@ -26,11 +26,16 @@ namespace eprosima {
 namespace fastdds {
 namespace dds {
 
+//! A enum with the possible values for the sample rejected reason
```
Comment (MiguelCompany) [inline include/fastdds/dds/core/status/SampleRejectedStatus.hpp:29]:
```suggestion
//! An enum with the possible values for the sample rejected reason
```

File: include/fastdds/dds/domain/DomainParticipantFactory.hpp [hunk 3]
```diff
@@ -88,23 +88,39 @@ class DomainParticipantFactory
...
 
-    //! Fills qos with the default values.
+    /**
+     * @brief This operation retrieves the default value of the DomainParticipant QoS, that is, the QoS policies which will
+     * be used for newly created DomainParticipant entities in the case where the QoS policies are defaulted in the
+     * create_participant operation.
+     * The values retrieved get_default_participant_qos will match the set of values specified  on the last successful call
+     * to set_default_participant_qos, or else, if the call was never made, the default values.
+     * @param qos DomainParticipantQos to be set
+     * @return RETCODE_INCONSISTENT_POLICY if the Qos is not self consistent and RETCODE_OK if the qos is changed correctly.
```
Comment (MiguelCompany) [inline include/fastdds/dds/domain/DomainParticipantFactory.hpp:111]: How could `RETCODE_INCONSISTENT_POLICY` be returned? Would that mean that the default qos is inconsistent?

File: include/fastdds/dds/domain/DomainParticipantFactory.hpp [hunk 4]
```diff
@@ -88,23 +88,39 @@ class DomainParticipantFactory
...
      */
     RTPS_DllAPI std::vector<DomainParticipant*> lookup_participants(
             DomainId_t domain_id) const;
 
-    //! Fills qos with the default values.
+    /**
+     * @brief This operation retrieves the default value of the DomainParticipant QoS, that is, the QoS policies which will
+     * be used for newly created DomainParticipant entities in the case where the QoS policies are defaulted in the
+     * create_participant operation.
+     * The values retrieved get_default_participant_qos will match the set of values specified  on the last successful call
```
Comment (MiguelCompany) [inline include/fastdds/dds/domain/DomainParticipantFactory.hpp:108]:
```suggestion
     * The values retrieved get_default_participant_qos will match the set of values specified on the last successful call
```

File: include/fastdds/dds/domain/DomainParticipantFactory.hpp [hunk 5]
```diff
@@ -88,23 +88,39 @@ class DomainParticipantFactory
...
+     * @return RETCODE_INCONSISTENT_POLICY if the Qos is not self consistent and RETCODE_OK if the qos is changed correctly.
+     */
     RTPS_DllAPI ReturnCode_t get_default_participant_qos(
             DomainParticipantQos& qos) const;
 
+    /**
+     * @brief This operation retrieves the default value of the DomainParticipant QoS, that is, the QoS policies which will
+     * be used for newly created DomainParticipant entities in the case where the QoS policies are defaulted in the
+     * create_participant operation.
+     * The values retrieved get_default_participant_qos will match the set of values specified  on the last successful call
```
Comment (MiguelCompany) [inline include/fastdds/dds/domain/DomainParticipantFactory.hpp:120]:
```suggestion
     * The values retrieved get_default_participant_qos will match the set of values specified on the last successful call
```

File: include/fastdds/dds/domain/DomainParticipantFactory.hpp (hunk 3 above)
Comment (laura-eprosima) [reply include/fastdds/dds/domain/DomainParticipantFactory.hpp:111]: This method returns `RETCODE_INCONSISTENT_POLICY` if the user data vector size exceeds the max_user_data limit

File: include/fastdds/dds/domain/DomainParticipantFactory.hpp [hunk 6]
```diff
@@ -117,12 +117,22 @@ class DomainParticipantFactory
...
     RTPS_DllAPI const DomainParticipantQos& get_default_participant_qos() const;
 
+    /**
+     * @brief This operation sets a default value of the DomainParticipant QoS policies which will be used for
+     * newly created DomainParticipant entities in the case where the QoS policies are defaulted in the
+     * create_participant operation.
+     *
+     * This operation will check that the resulting policies are self consistent; if they are not, the operation
+     * will have no effect and return INCONSISTENT_POLICY.
+     * @param qos DomainParticipantQos to be set
```
Comment (MiguelCompany) [inline include/fastdds/dds/domain/DomainParticipantFactory.hpp:133]: Explain the effect of passing `PARTICIPANT_QOS_DEFAULT` on this parameter.

Comment (laura-eprosima) [reply include/fastdds/dds/domain/DomainParticipantFactory.hpp:133]: It just set the `PARTICIPANT_QOS_DEFAULT` value to the default _participant_qos

File: include/fastdds/dds/core/policy/QosPolicies.hpp [hunk 7]
```diff
@@ -952,9 +952,11 @@ public:
                                                                                        \
     };
 
-
+/// @copydoc GenericDataQosPolicy
 TEMPLATE_DATA_QOS_POLICY(UserDataQosPolicy, PID_USER_DATA)
+/// @copydoc GenericDataQosPolicy
 TEMPLATE_DATA_QOS_POLICY(TopicDataQosPolicy, PID_TOPIC_DATA)
+/// @copydoc GenericDataQosPolicy
 TEMPLATE_DATA_QOS_POLICY(GroupDataQosPolicy, PID_GROUP_DATA)
```
Comment (EduPonz) [inline include/fastdds/dds/core/policy/QosPolicies.hpp:960]: Taking @MiguelCompany 's suggestion, I've tested this locally for Fast-RTPS-docs succesfully. The idea is to have a variable that is never defined anywhere, and then use the doxigen configuration file to set it to 1 in the PREDEFINED directive. This way, doxygen can see a class definition and it creates de corresponding XML file. As for the variable name, I just chose something, feel free to call it what you think is best and I'll adjust to that.
```
#ifdef DOXYGEN_DOCUMENTATION
/// @copydoc GenericDataQosPolicy
class UserDataQosPolicy {};
/// @copydoc GenericDataQosPolicy
class TopicDataQosPolicy {};
/// @copydoc GenericDataQosPolicy
class GroupDataQosPolicy {};
#endif

TEMPLATE_DATA_QOS_POLICY(UserDataQosPolicy, PID_USER_DATA)
TEMPLATE_DATA_QOS_POLICY(TopicDataQosPolicy, PID_TOPIC_DATA)
TEMPLATE_DATA_QOS_POLICY(GroupDataQosPolicy, PID_GROUP_DATA)
```

Comment (EduPonz) [reply include/fastdds/dds/core/policy/QosPolicies.hpp:960]: You can check https://github.com/eProsima/Fast-RTPS-docs/pull/117

File: include/fastdds/dds/publisher/DataWriter.hpp [hunk 8]
```diff
@@ -113,73 +109,86 @@ class DataWriter : public DomainEntity
      * Write data with handle.
      * @param data Pointer to the data
      * @param handle InstanceHandle_t.
-     * @return True if correct
-     * @par Calling example:
-     * @snippet fastrtps_example.cpp ex_PublisherWrite
+     * @return RETCODE_PRECONDITION_NOT_MET if the handle introduced does not match with the one associated to the data,
```
Comment (MiguelCompany) [inline include/fastdds/dds/publisher/DataWriter.hpp:112]: I think we should talk about the special value `HANDLE_NIL`

File: include/fastdds/dds/publisher/DataWriter.hpp [hunk 9]
```diff
@@ -113,73 +109,86 @@ class DataWriter : public DomainEntity
...
+     * RETCODE_OK if the data is correctly sent and RETCODE_ERROR otherwise.
```
Comment (MiguelCompany) [inline include/fastdds/dds/publisher/DataWriter.hpp:113]: Maybe it is not yet in the implementation, but this method could return TIMEOUT on some cases

File: include/fastdds/dds/publisher/DataWriter.hpp [hunk 10]
```diff
@@ -90,9 +90,7 @@ class DataWriter : public DomainEntity
     /**
      * Write data to the topic.
      * @param data Pointer to the data
-     * @return True if correct
-     * @par Calling example:
-     * @snippet fastrtps_example.cpp ex_PublisherWrite
+     * @return True if correct, false otherwise
      */
     RTPS_DllAPI bool write(
             void* data);
```
Comment (MiguelCompany) [inline include/fastdds/dds/publisher/DataWriter.hpp:96]: Should we change the return type of this to `ReturnCode_t` (perhaps on a different PR)?

File: include/fastdds/dds/publisher/DataWriter.hpp [hunk 11]
```diff
@@ -101,9 +99,7 @@ class DataWriter : public DomainEntity
      * Write data with params to the topic.
      * @param data Pointer to the data
      * @param params Extra write parameters.
-     * @return True if correct
-     * @par Calling example:
-     * @snippet fastrtps_example.cpp ex_PublisherWrite
+     * @return True if correct, false otherwise
      */
     RTPS_DllAPI bool write(
             void* data,
```
Comment (MiguelCompany) [inline include/fastdds/dds/publisher/DataWriter.hpp:105]: Should we change the return type of this to `ReturnCode_t` (perhaps on a different PR)?

File: include/fastdds/dds/publisher/DataWriter.hpp [hunk 12]
```diff
@@ -113,73 +109,86 @@ class DataWriter : public DomainEntity
...
-     * @return Topic data type
+     * Get data type associated to the DataWriter
+     * @return Copy of the TypeSupport
      */
     RTPS_DllAPI TypeSupport get_type() const;
 
     /**
      * Waits the current thread until all writers have received their acknowledgments.
+     * @param max_wait Maximum blocking time for this operation
+     * @return RETCODE_OK if the DataWriter receive the acknowledgments before the time expires and RETCODE_ERROR otherwise
```
Comment (MiguelCompany) [inline include/fastdds/dds/publisher/DataWriter.hpp:140]: What about RETCODE_TIMEOUT?

File: include/fastdds/dds/publisher/DataWriter.hpp [hunk 13]
```diff
@@ -190,13 +199,44 @@ class DataWriter : public DomainEntity
...
+     * operations on the DataReader objects that already knew that instance. This operation does not modify the value of
+     * the instance. The instance parameter is passed just for the purposes of identifying the instance.
+     * When this operation is used, the Service will automatically supply the value of the source_timestamp that is made
+     * available to DataReader objects by means of the source_timestamp attribute inside the SampleInfo. The constraints
+     * on the values of the handle parameter and the corresponding error behavior are the same specified for the
+     * unregister_instance operation.
+     * @param data Pointer to data.
+     * @return true if disposed, false if not.
+     */
     RTPS_DllAPI bool dispose(
```
Comment (MiguelCompany) [inline include/fastdds/dds/publisher/DataWriter.hpp:232]: Should we change the return type of this to `ReturnCode_t` (perhaps on a different PR)?

File: include/fastdds/dds/publisher/Publisher.hpp [hunk 14]
```diff
@@ -152,7 +158,9 @@ class Publisher : public DomainEntity
      * The deletion of the DataWriter will automatically unregister all instances.
      * Depending on the settings of the WRITER_DATA_LIFECYCLE QosPolicy, the deletion of the DataWriter
      * may also dispose all instances.
-     * @param writer
+     * @param writer DataWriter to delete
+     * @return RETCODE_PRECONDITION_NOT_MET if it not belongs to this Publisher, RETCODE_OK if it is correctly deleted and
```
Comment (MiguelCompany) [inline include/fastdds/dds/publisher/Publisher.hpp:162]:
```suggestion
     * @return RETCODE_PRECONDITION_NOT_MET if it does not belong to this Publisher, RETCODE_OK if it is correctly deleted and
```

File: include/fastdds/rtps/writer/ReaderProxy.h [hunk 15]
```diff
@@ -448,6 +448,7 @@ class ReaderProxy
     /**
      * @brief Find a change with the specified sequence number.
      * @param seq_num Sequence number to find.
+     * @param exact
```
Comment (MiguelCompany) [inline include/fastdds/rtps/writer/ReaderProxy.h:451]:
```suggestion
     * @param exact When false, the first change with a sequence number not less than seq_num will be returned.
                    When true, the change with a sequence number value of seq_num will be returned.
```

File: include/fastdds/dds/log/Log.hpp [hunk 16]
```diff
@@ -28,20 +28,20 @@
...
 // Logging API:
 
-//! Logs an info message. Disable it through Log::SetVerbosity, #define LOG_NO_INFO, or being in a release branch
+//! Logs an info message. Disable it through Log::SetVerbosity, define LOG_NO_INFO, or being in a release branch
 #define logInfo(cat, msg) logInfo_(cat, msg)
-//! Logs a warning. Disable reporting through Log::SetVerbosity or #define LOG_NO_WARNING
+//! Logs a warning. Disable reporting through Log::SetVerbosity or define LOG_NO_WARNING
 #define logWarning(cat, msg) logWarning_(cat, msg)
-//! Logs an error. Disable reporting through #define LOG_NO_ERROR
+//! Logs an error. Disable reporting through define LOG_NO_ERROR
```
[the remark this answers is not in the document]
Comment (laura-eprosima) [reply include/fastdds/dds/log/Log.hpp:44]: Because it causes a warning in Doxygen

File: include/fastdds/dds/publisher/DataWriter.hpp (hunk 10 above)
Comment (laura-eprosima) [reply include/fastdds/dds/publisher/DataWriter.hpp:96]: It is a non-standard method, but if you want to change the return type I think it is better to do it in a separate PR

File: include/fastdds/dds/core/policy/QosPolicies.hpp [hunk 17]
```diff
@@ -780,26 +952,79 @@ public:
...
+ * @brief Class derived from GenericDataQosPolicy
+ *
+ * The purpose of this QoS is to allow the application to attach additional information to the created
+ * Entity objects such that when a remote application discovers their existence it can access that information and
+ * use it for its own purposes.
+ *
+ * One possible use of this QoS is to attach security credentials or some other information that can be used by the
+ * remote application to authenticate the source.
+ */
+class UserDataQosPolicy
```
Comment (MiguelCompany) [inline include/fastdds/dds/core/policy/QosPolicies.hpp:967]:
```suggestion
class UserDataQosPolicy : public GenericDataQosPolicy
```

File: include/fastdds/dds/core/policy/QosPolicies.hpp [hunk 18]
```diff
@@ -780,26 +952,79 @@ public:
...
+ * @brief Class derived from GenericDataQosPolicy
+ *
+ * The purpose of this QoS is to allow the application to attach additional information to the created Topic
+ * such that when a remote application discovers their existence it can examine the information and use it in an
+ * application-defined way.
+ *
+ * In combination with the listeners on the DataReader and DataWriter as well as by means of operations such as
+ * ignore_topic,these QoS can assist an application to extend the provided QoS.
+ */
+class TopicDataQosPolicy
```
Comment (MiguelCompany) [inline include/fastdds/dds/core/policy/QosPolicies.hpp:980]:
```suggestion
class TopicDataQosPolicy : public GenericDataQosPolicy
```

File: include/fastdds/dds/core/policy/QosPolicies.hpp [hunk 19]
```diff
@@ -780,26 +952,79 @@ public:
...
+ *
+ * The purpose of this QoS is to allow the application to attach additional information to the created
+ * Publisher or Subscriber. The value of the GROUP_DATA is available to the application on the DataReader and
+ * DataWriter entities and is propagated by means of the built-in topics.
+ *
+ * This QoS can be used by an application combination with the DataReaderListener and DataWriterListener to
+ * implement matching policies similar to those of the PARTITION QoS except the decision can be made based on an
+ * application-defined policy.
+ */
+class GroupDataQosPolicy
```
Comment (MiguelCompany) [inline include/fastdds/dds/core/policy/QosPolicies.hpp:994]:
```suggestion
class GroupDataQosPolicy : public GenericDataQosPolicy
```

File: include/fastdds/dds/core/policy/QosPolicies.hpp [hunk 20]
```diff
@@ -1068,33 +1366,57 @@ class PartitionQosPolicy : public Parameter_t, public QosPolicy
         return *this;
     }
 
+    /**
+     * @brief Getter for the first position of the ParameterPropertyList
```
Comment (MiguelCompany) [inline include/fastdds/dds/core/policy/QosPolicies.hpp:1370]:
```suggestion
     * @brief Getter for the first position of the partition list
```

File: include/fastdds/dds/core/policy/QosPolicies.hpp [hunk 21]
```diff
@@ -1068,33 +1366,57 @@ class PartitionQosPolicy : public Parameter_t, public QosPolicy
...
+     * @return const_iterator
+     */
     const_iterator begin() const
     {
         return const_iterator(partitions_.data);
     }
 
+    /**
+     * @brief Getter for the end of the ParameterPropertyList
```
Comment (MiguelCompany) [inline include/fastdds/dds/core/policy/QosPolicies.hpp:1379]:
```suggestion
     * @brief Getter for the end of the partition list
```

File: include/fastdds/dds/core/policy/QosPolicies.hpp [hunk 22]
```diff
@@ -1300,19 +1676,18 @@ class ResourceLimitsQosPolicy : public Parameter_t, public QosPolicy
...
- * Class DurabilityServiceQosPolicy, to indicate the Durability Service.
- * This QosPolicy can be defined and is transmitted to the rest of the network but is not implemented in this version.
- * service_cleanup_delay: Default value c_TimeZero.
- * history_kind: Default value KEEP_LAST_HISTORY_QOS.
- * history_depth: Default value 1.
- * max_samples: Default value -1.
- * max_instances: Default value -1.
- * max_samples_per_instance: Default value -1.
+ * Specifies the configuration of the durability service. That is, the service that implements the DurabilityQosPolicy kind
+ * of TRANSIENT and PERSISTENT.
```
Comment (MiguelCompany) [inline include/fastdds/dds/core/policy/QosPolicies.hpp:1680]:
```suggestion
 * of PERSISTENT.
```

File: include/fastdds/dds/publisher/DataWriter.hpp (hunk 12 above)
Comment (laura-eprosima) [reply include/fastdds/dds/publisher/DataWriter.hpp:140]: Even if the DDS Standard states that this method should return RETCODE_TIMEOUT, that situation is not differentiated by now.

File: include/fastdds/dds/publisher/DataWriter.hpp (hunk 9 above)
Comment (laura-eprosima) [reply include/fastdds/dds/publisher/DataWriter.hpp:113]: That ReturnCode is not returned by the moment

File: include/fastdds/dds/core/policy/QosPolicies.hpp [hunk 23]
```diff
@@ -342,8 +452,14 @@ class LivelinessQosPolicy : public Parameter_t, public QosPolicy
...
 public:
 
+    //!Liveliness kind <br> By default, AUTOMATIC_LIVELINESS.
     LivelinessQosPolicyKind kind;
+    //! Indicates for how much time should a remote participant consider the local participant to be alive. <br>
+    //! By default, c_TimeInfinite.
     fastrtps::Duration_t lease_duration;
+    //! The period for the participant to send PDP announcements.
+    //! By default, c_TimeInfinite.
+    //! @warning Must be < lease_duration.
```
[the remark this answers is not in the document]
Comment (laura-eprosima) [reply include/fastdds/dds/core/policy/QosPolicies.hpp:462]: The announcement_period comment was already there. What do you think is a better definition for that parameter?

File: include/fastdds/dds/core/policy/QosPolicies.hpp (hunk 22 above)
Comment (laura-eprosima) [reply include/fastdds/dds/core/policy/QosPolicies.hpp:1680]: In the DDS Standard states that this QoS Policy is applied if the Durability is either TRANSIENT or PERSISTENT.

File: include/fastdds/dds/core/status/SampleRejectedStatus.hpp [hunk 24]
```diff
@@ -26,11 +26,16 @@ namespace eprosima {
...
 enum SampleRejectedStatusKind
 {
+    //!Default value
     NOT_REJECTED,
+    //! Exceeds the max_instance limit
     REJECTED_BY_INSTANCES_LIMIT,
+    //! Exceeds the max_samples limit
     REJECTED_BY_SAMPLES_LIMIT,
+    //! Exceeds the max_samples_per_instance limit
     REJECTED_BY_SAMPELS_PER_INSTANCE_LIMIT
```
[the remark this answers is not in the document]
Comment (MiguelCompany) [reply include/fastdds/dds/core/status/SampleRejectedStatus.hpp]: This has not yet been changed

[the remark this answers is not in the document]
Comment (MiguelCompany) [reply include/fastdds/dds/core/status/SampleRejectedStatus.hpp]: Forget it. The link took me to a wrong place.

**Pull request merged**